### ISSUE TITLE
A doubt about ProxylessNAS

### ISSUE THREAD

**Issue by AliCloud-PAI** [microsoft/nni#3519]
Hi, @ultmaster  ~
At below code line:
nni/algorithms/nas/pytorch/proxylessnas/trainer.py#L150

Should the mode of model be in eval() mode under validation? 
Thanks :)

**Comment by ultmaster** [comment]
I think setting to training mode when validation saves the extra cost of Batch-norm calibration (or setting `track_running_stats` to False). You can refer to papers like SPOS or Once-for-all for why people calibrate batch-norm on supernet. There is still debate on whether this is essential, as some algorithms report good performance even without this trick on some search spaces.

Note that this version of ProxylessNAS isn't maintained any more. Please refer to the Retiarii version. As far as I remember, Retiarii respects the original batch norm statistics when evaluation (or maybe I haven't implemented the evaluation at all).

**Comment by AliCloud-PAI** [comment]
> I think setting to training mode when validation saves the extra cost of Batch-norm calibration (or setting `track_running_stats` to False). You can refer to papers like SPOS or Once-for-all for why people calibrate batch-norm on supernet. There is still debate on whether this is essential, as some algorithms report good performance even without this trick on some search spaces.
> 
> Note that this version of ProxylessNAS isn't maintained any more. Please refer to the Retiarii version. As far as I remember, Retiarii respects the original batch norm statistics when evaluation (or maybe I haven't implemented the evaluation at all).
> 
> Involving @QuanluZhang as he probably knows better than me.

I see that the validation is not used in Proxylessnas of Retiarii. 
If we want to use the pre-trained parameters of the supernet for fine-tuning of sampled arch, the train mode may not be suitable under validation? 
Thanks:)

**Comment by ultmaster** [comment]
As I said, you need calibrate BN. If you didn't calibrate BN, I suspect that training mode is actually better than validation mode.

**Comment by AliCloud-PAI** [comment]
> As I said, you need calibrate BN. If you didn't calibrate BN, I suspect that training mode is actually better than validation mode.

Thanks a lot:)
